crates/mir/src/emit/emitter.rs

```diff
@@ -93,7 +111,7 @@ impl MirEmitter {
     /// # Safety
     /// This function is unsafe because it returns a [`VariableId`] which does not have to be properly registered in
     /// scope, what can cause a compiler or runtime panic. The caller must ensure that the [`VariableId`] is properly.
-    pub(crate) unsafe fn next_value_id(&mut self) -> ValueId {
+    unsafe fn next_value_id(&mut self) -> ValueId {
         let variable_id = mem::transmute(self.next_value_id);
         self.next_value_id += 1;
         variable_id
```

Comment:
This on the other hand is a good use of `unsafe` :smile: 

Using this function directly can break invariants, which is exactly what `unsafe` was built for. I just don't like the usage of `std::mem::transmute` inside. It doesn't seem necessary for what's being done here; I'd argue we should be avoiding unsafe code even if it means writing a few more lines.